components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsWrapperFactorySingleton.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.config.model.graph;
+
+import org.wso2.carbon.identity.application.authentication.framework.internal.FrameworkServiceDataHolder;
+
+/**
+ * Singleton to get the appropriate Javascript Proxy Object Wrapper Factory.
+ */
+public class JsWrapperFactorySingleton {
+
+    private static JsWrapperFactorySingleton jsWrapperFactorySingleton = null;
+    private static JsWrapperFactory jsWrapperFactory = null;
+
+
+    private JsWrapperFactorySingleton() {
+        if (FrameworkServiceDataHolder.getInstance().
+                getJsGraphBuilderFactory() instanceof JsPolyglotGraphBuilderFactory) {
+            jsWrapperFactory = new GraalJsWrapperFactory();
+        } else {
+            jsWrapperFactory = new NashornJsWrapperFactory();
+        }
+    }
+
+    public static JsWrapperFactory getInstance() {
+        if (jsWrapperFactorySingleton == null) {
+            jsWrapperFactorySingleton = new JsWrapperFactorySingleton();
+        }
+        return jsWrapperFactory;
+    }
+
+    public static void clearInstance() {
```

Comment:
No need to clear the instance.

Reply:
This is required while running the tests